### PR TITLE
Feat rspy148 149 158/client classes julien

### PR DESCRIPTION
**Pull request by jgaucher-cs**

### CODE REVIEW

File: rs_workflows/staging.py [hunk 1]
```diff
@@ -580,16 +356,16 @@ def __init__(  # pylint: disable=too-many-arguments
...
         """
-        super().__init__(user, url, url_catalog, station, mission, tmp_download_path, s3_path, apikey)
+        super().__init__(rs_client, mission, tmp_download_path, s3_path)
         self.max_workers: int = max_workers
         self.start_datetime: datetime = start_datetime
         self.stop_datetime: datetime = stop_datetime
         self.limit = limit
 
 
 @flow(task_runner=DaskTaskRunner(cluster_kwargs={"n_workers": 15, "threads_per_worker": 1}))
```
Comment (jgaucher-cs) [inline rs_workflows/staging.py:366]: @agrosu1978 it seems to me here that we have the same information twice: the number of workers passed to prefect (the `n_workers` parameter) is the same than `PrefectFlowConfig.max_workers` ? 

So maybe we could remove `PrefectFlowConfig.max_workers` and take the information from `n_workers` instead ? (if we can retrieve this parameter value from the python code ?)

File: rs_workflows/s1_l0.py [hunk 2]
```diff
@@ -430,23 +413,23 @@ def s1_l0_flow(config: PrefectS1L0FlowConfig):
...
+        **RsClient.create_apikey_headers(config.apikey),
     )
     fin_res = []
     for output_product in get_yaml_outputs(yaml_dpr_input):
         matching_stac = next(
             (d for d in files_stac.result() if d["stac_discovery"]["properties"]["eopf:type"] in output_product),
             None,
         )
-        # To be removed, temp fix
+        # To be removed, temp fix (?)
```
Comment (jgaucher-cs) [inline rs_workflows/s1_l0.py:424]: idem here @agrosu1978 @Padeanu what needs to be fixed ?

File: tests/test_s1_l0.py [hunk 3]
```diff
@@ -371,18 +375,20 @@ def test_get_adgs_catalog_data(endpoint, status):
...
             assert adgs_res == adgs_catalog
         else:
             assert adgs_res is None
     else:
-        assert get_adgs_catalog_data.fn(endpoint, username, collection, files_list, apikey) is None
+        assert adgs_res is None
 
 
-# TODO: The unit testing for this prefect flow does not work
+# TODO: The unit testing for this prefect flow does not work (?)
```
Comment (jgaucher-cs) [inline tests/test_s1_l0.py:391]: @agrosu1978 you need to fix that test before merging to develop ?

File: rs_client/cadip_client.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+    @property
+    def href_status(self) -> str:
+        """Return the RS-Server hostname and path where the CADIP status endpoint is deployed."""
+        return f"{self.href_cadip}/cadip/{self.station.value}/cadu/status"
+
+    @property
+    def station_name(self) -> str:
+        """Return the station name."""
+        return f"CADIP/{self.station.value}"
```
Comment (jgaucher-cs) [inline rs_client/cadip_client.py:72]: @agrosu1978 @Padeanu it's a detail but we could just return "CADIP" here

File: rs_workflows/staging.py (hunk 1 above)
Comment (agrosu1978) [reply rs_workflows/staging.py:357]: Prefect creates a temporary local cluster automatically with the number of workers used based on the number of cores on your machine. I don't know how we are going to use the Dask cluster on our cluster, is this going to be used locally on the pod, or a specific one deployed on our cluster? So, the n_workers is the maximum number of tasks that can be running in parallel. The max_workers is the number of tasks that our flow can create, but if this is higher than n_workers, of course that only n_workers will run in parallel, while the (max_workers - n_workers) tasks wait for a running tasks to finish in order to take that slot.
I noticed that using n_workers with a high value, simply blocks the PC. Anyway, we can see in the future after some experimentation with prefect flows if in fact the max_workers may be the same as n_workers. Let's keep it like this for now

File: rs_workflows/s1_l0.py [hunk 5]
```diff
@@ -416,7 +399,7 @@ def s1_l0_flow(config: PrefectS1L0FlowConfig):
         logger.error("DPR did not processed anything")
         return
 
-    # Temp, to be fixed
+    # Temp, to be fixed (?)
```
[the remark this answers is not in the document]
Comment (agrosu1978) [reply rs_workflows/s1_l0.py:417]: I was referring at the name of the collection. As yo can see, I assumed that the name of the collection where the final results are to be published is "s1_dpr" (in the case of s1 mission). But maybe this collection should be an input from the user in fact? What if he wants to name that collection with a specific name? There is no guide in the US for this

File: tests/test_s1_l0.py (hunk 3 above)
Comment (agrosu1978) [reply tests/test_s1_l0.py:405]: No, I will remove it. All the functions used in the flow are tested with the others tests.

File: rs_workflows/staging.py [hunk 6]
```diff
@@ -0,0 +1,437 @@
...
+    Attributes:
+        rs_client (AuxipClient | CadipClient): RsClient instance
+        mission (str): The mission identifier.
+        tmp_download_path (str): The temporary download path.
+        s3_path (str): The S3 path for storing downloaded files.
+    """
+
+    def __init__(
+        self,
+        rs_client: AuxipClient | CadipClient,
```
[the remark this answers is not in the document]
Comment (agrosu1978) [reply rs_workflows/staging.py:118]: The decision to use the RSClient object as input for the prefect flow has been taken. Consider this comment as solved

File: rs_workflows/staging.py [hunk 7]
```diff
@@ -0,0 +1,437 @@
...
+        Initialize the PrefectTaskConfig object with provided parameters.
+        """
+
+        super().__init__(rs_client, mission, tmp_download_path, s3_path)
+        self.task_files_stac: list[dict] = task_files_stac
+        self.max_retries: int = max_retries
+
+
+@task
+def staging(config: PrefectTaskConfig):
```
[the remark this answers is not in the document]
Comment (agrosu1978) [reply rs_workflows/staging.py:162]: We can have some checks in the future if errors appear, consider this comment as solved for now

**Review by agrosu1978 — CHANGES_REQUESTED** [review]
Implement the inheritance from RsClient for Cadip/AuxipClient

**Pull request merged**